addons/File_Sender/src/presenter.js

```diff
@@ -0,0 +1,699 @@
+            if(scaleInfo.scaleY!==1.0) {
+                $dialog.css('transform', scaleInfo.transform);
+                $dialog.css('transform-origin', scaleInfo.transformOrigin);
+            }
+
+            if(parseFloat(window.MobileUtils.getAndroidVersion())=='4.1'){
+                if (window !== window.top) {
+                    var ancestorData;
+                    for (var i=0; i<presenter.ancestorsData.length; i++)
+                    {
+                        ancestorData = presenter.ancestorsData[i];
+                        $(ancestorData.wnd).scrollTop(ancestorData.offset);
+                    }
+                    presenter.ancestorsData = undefined;
+                }
+            }
+
+            if (isPopup || presenter.isPreview) {
+                popupLeft = presentationPosition.left;
+                topPosition = parseInt((availableHeight - dialogHeight) / 2, 10);
+            }
+            else {
+                topPosition = parseInt(( windowHeight - dialogHeight) / 2, 10) ;
+            }
+
+            var presentationHorizontalOffset = parseInt((presentationWidth - dialogWidth) * scaleInfo.scaleY / 2, 10);
+            var leftPosition = presentationPosition.left + presentationHorizontalOffset;
+
+            // adjust top position if Player was embedded in iframe (i.e. EverTeach)
+            if (window !== window.top) {
+                var iframe = window.parent.document.getElementsByTagName('iframe');
+                var iframeDialogHeight = parseInt($dialog.height(), 10);
+                iframeDialogHeight += DOMOperationsUtils.calculateOuterDistances(DOMOperationsUtils.getOuterDimensions($dialog)).vertical;
+
+                if (topPosition < 0) {
+                    topPosition = 0;
+                } else if (topPosition > $(window).height() - iframeDialogHeight) {
+                    topPosition = $(window).height() - iframeDialogHeight;
+                }
+            }
+
+
+            if ($(window).scrollTop() > popupTop && isPopup) {
+                topPosition += ($(window).scrollTop() - popupTop);
+            }
+
+            $dialog.css({
+                left: (leftPosition - popupLeft) + 'px',
+                top: (topPosition + scrollTop + previewFrame) + 'px'
+            });
+
+            $dialog.find('.ui-dialog-content').css({
+                color: 'black'
+            });
+
+            if(isPopup || presenter.isPreview) {
+                // For Preview and Popup dialog is moved to appropriate page
+                var $overlay = $(".ui-widget-overlay");
+                $(presenter.$view.closest(".ui-widget-overlay")).remove();
+                if (isPreview) {
+                    $(".ic_page_panel").children(".ic_page").children().last().after($overlay);
+                }
+                else {
+                    $dialog.before($overlay);
+                }
+            }
+        }catch(e){console.log(e)}
```

Comment:
Dobrze byłoby trochę rozbić to co dzieje się w części try na podfunkcje, ze względu na to, że to wielka ściana kodu. W części catch można zastosować console.error